app/src/main/java/com/patrickgatewood/weather/ui/WeatherActivity.java

```diff
@@ -43,24 +38,33 @@ protected void onCreate(Bundle savedInstanceState) {
         ((WeatherApplication) getApplication()).getApplicationComponent().inject(this);
         ButterKnife.bind(this);
 
-        weatherViewModel.addObserver(this);
+        weatherPresenter.attachView(this);
         fetchApiDataButton.setOnClickListener(new View.OnClickListener() {
             @Override
             public void onClick(View v) {
-                weatherViewModel.onFetchButtonTap();
+                weatherPresenter.onFetchButtonClick();
             }
         });
     }
 
     @Override
-    public void update(Observable observable, Object arg) {
-        Log.v(TAG, "Update called");
-
-        if (observable instanceof WeatherViewModel)     {
-            Log.v(TAG, "Received correct updates from viewmodel");
-            WeatherViewModel weatherViewModel = (WeatherViewModel) observable;
-            ForecastData currentForecastData = weatherViewModel.getCurrentForecast().getCurrentForecastData();
-            summaryTextView.setText(currentForecastData.getSummary());
-        }
+    protected void onResume() {
+        super.onResume();
+    }
+
+    @Override
+    protected void onDestroy() {
+        weatherPresenter.detachView();
+        super.onDestroy();
+    }
+
+    @Override
+    public void updateTextViews(String temperature, String feelsLikeTemp, String summary) {
```

Comment:
TextViews? What kind of textviews? Why are they there? What are they doing? Might want to add more information in the method name.